src/bucket.ts

```diff
@@ -3454,7 +3454,10 @@ class Bucket extends ServiceObject {
           return;
         }
 
-        options.resumable = fd.size > RESUMABLE_THRESHOLD;
+        if (fd.size <= RESUMABLE_THRESHOLD) {
+          // Only disable resumable uploads so createWriteStream still attempts them and falls back to simple upload.
```

Comment:
Doesn't this have the same effect as the removed code?

Reply:
It doesn't explicitly set `resumable` to `true` so when the below check is done (Line 1723), it will fallback to a simple upload (I've mentioned this in more detail in [#1133](https://github.com/googleapis/nodejs-storage/issues/1133#issuecomment-605323416)

https://github.com/googleapis/nodejs-storage/blob/6833dfe9bf013a582ff46620b8035fc47ec38d0f/src/file.ts#L1721-L1741

Reply:
Sorry, I'm caught up now :)

This change would ultimately opt the user out of resumable upload behavior by default when they pass a large file. That would mean the threshold concept wouldn't have a purpose, but it has been working for us so far.

The error returned about the directory that isn't writable isn't a retryable error, since resumable upload relies on using that directory to persist data that allow it to resume the upload if it fails during transmission. The message suggests using "resumable: false", but that's only a suggestion, as the user may prefer to resolve the error and gain the benefits of a resumable upload.

Reply:
>This change would ultimately opt the user out of resumable upload behavior by default when they pass a large file

It wouldn't - from my understanding, passing `resumable=false` to `File.createWriteStream` would be the only thing which prevents a resumable upload? `File.createWriteStream` always tries to do a resumable upload unless `resumable` is `false`.

This change is just so that `Bucket.upload` performs the same as `File.createWriteStream` when no `resumable` option is passed to `File.createWriteStream`. `File.createWriteStream` currently ignores the error returned about the directory that isn't writable and continues to a simple upload, shouldn't `Bucket.upload` match this?

Reply:
> shouldn't Bucket.upload match this?

That's a fair question. `createWriteStream()` defaults to resumable because it's the best-case scenario for a user. But, it doesn't know the incoming data size, so if it turns out a resumable upload doesn't work, then "oh well, maybe we were wrong and the file isn't that big anyway" and a non-resumable upload is fine. With `bucket.upload()`, we know the file size, and if it's large enough that we think resumable would be best, we opt the user into it, as opposed to being passive about it.

I could go either way here, I think. It's nice to just make things work without obstructions or annoying errors, but it's hard to know if we would be disappointing other users who would prefer to have the perks of resumable uploads, and appreciate the error.

@jkwlui @frankyn @bcoe any thoughts?

Reply:
I am leaning towards matching `bucket.upload()`'s behaviour with `file.createWriteStream()`'s so we reduce the complexity of different upload strategies, but @stephenplusplus's point about informing user about the advantage of resumable uploads is also very valid - I wonder if better documentation about the benefit of explicitly opting resumable upload and having the user check that the config dir is writable in their environment would be a fair middle ground here. 

Nevertheless, we should probably align our libraries strategies across languages. Summarizing the current behaviour for those who's just getting caught up with this change:

For Bucket.upload() where the file size is known in advance, we default resumable upload to true if `size > RESUMABLE_THRESHOLD`, which will result in error if the config dir is not writable (like in Cloud Run as pointed out by @rhodgkins ).

Whereas for `file.createWriteStream()` where a user is expected to pipe a readable stream to and the file size is *not* known in advance, we default to simple upload if not specified.

@frankyn

Reply:
> Whereas for file.createWriteStream() where a user is expected to pipe a readable stream to and the file size is not known in advance, we default to simple upload if not specified.

We actually default to resumable, but just silently fallback to a simple upload if it didn't work.

Reply:
Thanks, edited the comment.